projects/js-packages/ai-client/src/hooks/use-media-recording/Readme.md

```diff
@@ -0,0 +1,48 @@
+# `useMediaRecording` Custom React Hook
+
+## Description
+
+`useMediaRecording` is a custom React hook for handling media recording functionalities in a React application. It provides an easy way to start, pause, resume, and stop media recording, as well as to track the current recording state.
+
+Based on [MediaRecorder](https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder) API
+
+## API
+
+The hook returns an object with the following properties and methods:
+
+- `start: () => void`: Start the media recording
+- `pause: () => void`: Pause the current media recording
+- `resume: () => void`: Resume a paused recording
+- `stop: () => void`: Stop the current recording
+- `state: 'inactive' | 'recording' | 'paused'`: Current recording state
+
+## Example
+
+Here's an example React component that utilizes the `useMediaRecording` hook.
+
+```jsx
+import useMediaRecording from './useMediaRecording';
+
+const MediaRecorderComponent = () => {
+	const { start, pause, resume, stop, state } = useMediaRecording();
+
+	return (
+		<div>
+			<h1>Media Recorder</h1>
+			<p>Current State: { state }</p>
+			<button onClick={ start } disabled={ state === 'recording' }>
```

Comment:
```suggestion
			<button onClick={ start } disabled={ state !== 'inactive' }>
```
If it is either recording or paused the `Start` should be disabled, right? Or else we could start again after pausing.